### DISCUSSION TITLE
Do resonance hybrids from SMARTS mols behave the same as mols from SMILES?

### DISCUSSION THREAD

**Discussion by ljmartin** [rdkit/rdkit#6135] [Q&A] [answered]
SMARTS molecules are drawn as resonance hybrids in the aromatic regions, with a dotted line as the second 'bond' that surrounds the whole aromatic region. Molecules from SMILES are drawn as a kekule form. 

Do the SMARTS mols have the same behaviour as a SMILES mol? 

As a quick check, I see that, for instance, bonds aren't classified as conjugated in the SMARTS mol:
```python
#phenol 
m = Chem.MolFromSmiles('c1ccc(O)cc1')
for bond in m.GetBonds():
    print(bond.GetIsConjugated())
```
(results are all `True`).
```python
m =Chem.MolFromSmarts(Chem.MolToSmarts(m))
for bond in m.GetBonds():
    print(bond.GetIsConjugated())
```
(results are all `False`). 

Is there an obvious origin for why this is? I'm trying to figure out what other differences there might be. 

Thanks
Lewis

**Comment by greglandrum** [comment, marked as answer]
Hi Lewis,
The main differences between molecules constructed from SMILES and SMARTS are that those from SMARTS contain QueryAtoms and QueryBonds, as opposed to Atoms and Bonds, and are not sanitized (sanitization is described here: https://www.rdkit.org/docs/RDKit_Book.html#molecular-sanitization). Conjugation is calculated as part of the sanitization process. Though you could technically ask the code to do the conjugation calculation for molecules with QueryAtoms and QueryBonds, this wouldn't really make much sense in most applications.

On the topic of drawing: The default behavior of the molecule drawing code is to kekulize molecules since, in my experience, chemists tend to prefer to kekulized drawings. The kekulized form in the drawing doesn't reflect the actual status of the molecule . The kekulization also does not modify QueryBonds which have bond type queries (like what you get in SMARTS).

**Comment by ljmartin** [comment]
TYSM Greg!
On this topic - is there some way to treat non-aromatic atoms in a resonance form as equivalent? For example, a charged carboxylate: I (chemically) expect the oxygens to be equivalent - so the charge is spread across the two oxygens, and they have something like a 1.5 bond to the carbon. 
My intended use-case is for featurization with Morgan bits, where this isn't the case (note the oxygens have different bits):
```python
from rdkit import Chem
from rdkit.Chem import AllChem

def mol_to_bits(mol, useFeatures=False):
    bi = {}
    fp = AllChem.GetMorganFingerprint(mol, 0, bitInfo=bi,useFeatures=useFeatures)
    bits = np.zeros(mol.GetNumAtoms(),dtype=int)
    for key, value in bi.items():
        for v in value:
            bits[v[0]] = key
    return bits

m = Chem.MolFromSmiles('C([O-])=O')
print('0th neighbourhood bits:', mol_to_bits(m))
```
output (carbon, oxygen, oxygen):
`0th neighbourhood bits: [2246703798  864942795  864942730]`

**Reply by greglandrum** [reply]
There's not currently any way to automatically do this. If you want to go down this road, then I would suggest writing a function which edits the molecule before you generate the fingerprint and does whatever charge/bond-order symmetrization you want to do.
